### PR TITLE
Add OAuth2 config options for credentials location

### PR DESCRIPTION
**Pull request by ekoleda-codaio**
Our current OAuth2 token exchange logic first tries to pass the client ID and secret in the request body, and if that fails retries it in an Authorization basic header. Some APIs (Airtable for example) require the Authorization header, but invalidate the authorization code after the first failed attempt in the body.

This PR adds a new option to the OAuth2 config that allows the developer to explicitly specify which method to use. A corresponding change will need to be made to the Coda server to honor this setting.

### CODE REVIEW

**Comment by huayang-codaio** [comment]
lgtm. let me send out a quick change reflecting this sdk option.

**Review by jonathan-codaio — APPROVED** [review]
👍

**Review by huayang-codaio — APPROVED** [review]
implemented with https://github.com/coda/coda/pull/82557

**Comment by ekoleda-codaio** [comment]
> if some oauth provider adopts difference practices for exchange v.s. refresh

I haven't seen that before in the wild, so fingers crossed it never happens 🤞

File: testing/upload_validation.ts [hunk 1]
```diff
@@ -424,6 +424,7 @@ const defaultAuthenticationValidators: Record<AuthenticationType, z.ZodTypeAny>
     pkceChallengeMethod: z.enum(['plain', 'S256']).optional(),
     scopeParamName: z.string().optional(),
     nestedResponseKey: z.string().optional(),
+    credentialsLocation: z.string().optional(),
```
Comment (jonathan-codaio) [inline testing/upload_validation.ts:427]: Oops should have caught this earlier, this should be `z.nativeEnum(TokenExchangeCredentialsLocation).optional()`

Comment (ekoleda-codaio) [reply testing/upload_validation.ts:427]: Fixed in https://github.com/coda/packs-sdk/pull/2354

**Pull request merged**